src/m365/pp/commands/card/card-remove.spec.ts

```diff
@@ -0,0 +1,223 @@
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(appInsights, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      promptOptions = options;
+      return { continue: false };
+    });
+    promptOptions = undefined;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.get,
+      request.delete,
+      powerPlatform.getDynamicsInstanceApiUrl,
+      Cli.prompt,
+      Cli.executeCommandWithOutput
+    ]);
+  });
+
+  after(() => {
+    sinonUtil.restore([
+      auth.restoreAuth,
+      appInsights.trackEvent,
+      pid.getProcessName
+    ]);
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.CARD_REMOVE), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if id is not a valid guid.', async () => {
+    const actual = await command.validate({
+      options: {
+        environment: validEnvironment,
+        id: 'Invalid GUID'
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (id)', async () => {
+    const actual = await command.validate({ options: { environment: validEnvironment, id: validId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (name)', async () => {
+    const actual = await command.validate({ options: { environment: validEnvironment, name: validName } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('prompts before removing the specified card owned by the currently signed-in user when confirm option not passed', async () => {
+    sinon.stub(powerPlatform, 'getDynamicsInstanceApiUrl').callsFake(async () => envUrl);
+
+    await command.action(logger, {
+      options: {
+        environment: validEnvironment,
+        id: validId
+      }
+    });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified card owned by the currently signed-in user when confirm option not passed and prompt not confirmed', async () => {
+    sinon.stub(powerPlatform, 'getDynamicsInstanceApiUrl').callsFake(async () => envUrl);
```

Comment:
@nicodecleyre seems like you completely removed this test which was not what I meant. This test is useful for sure.
I just wanted to say that this `sinon.stub` could be removed because we're not executing any logic (we abort the prompt).